### PR TITLE
Gift Entry Payment Services: Hide Payment Services widget when not an Elevate customer

### PR DESCRIPTION
**Pull request by kenlewis-sfdo**
This set of changes hides the payment services widget from the Template Builder sidebar if an org is not an Elevate customer.  It defines an Elevate customer by having the following configuration values set in the org:

elevateSDK, productId, baseURL, apiKey, sfdo.merchantids, sfdo.gatewayids

These configuration keys are required in order to load the Elevate SDK and to make callouts to Elevate services.

[W-040100](https://foundation.lightning.force.com/lightning/r/agf__ADM_Work__c/a2x1E000001HtRpQAK/view)

# Critical Changes

# Changes

# Issues Closed

# Community Ideas Delivered

# New Metadata

# Deleted Metadata

### CODE REVIEW

**Comment by kenlewis-sfdo** [comment]
**lurch attach: W-040100

File: src/classes/GE_PaymentServices.cls [hunk 1]
```diff
@@ -139,8 +148,8 @@ public with sharing class GE_PaymentServices {
     public with sharing class JWTPayload {
         String sfdoId;
 
-        public JWTPayload(GE_PaymentServices serviceInstance) {
-            sfdoId = serviceInstance.config.get(serviceInstance.PRODUCTID);
+        public JWTPayload(String sfdoId) {
```
Comment (ptewson-sfdo) [inline src/classes/GE_PaymentServices.cls:151]: Why is the sfdoId the one parameter we are passing to the class?

Comment (kenlewis-sfdo) [reply src/classes/GE_PaymentServices.cls:151]: Right now this is the only config value that this class uses.  If we eventually read product or roles from the passed-in configuration then we can add those params to the constructor.

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x1E000001HtRpQAK" target="_blank">W-040100</a>

File: src/classes/GE_PaymentServices_TEST.cls [hunk 2]
```diff
@@ -180,8 +180,35 @@ public class GE_PaymentServices_TEST {
...
+        for (Integer i = 0; i < GE_PaymentServices.REQUIRED_CONFIG_KEYS.size(); i++) {
+            requiredConfigs.add(new Payment_Services_Configuration__c(
+                    Service__c = GE_PaymentServices.PAYMENTS_SERVICE_NAME,
+                    Key__c = GE_PaymentServices.REQUIRED_CONFIG_KEYS.get(i),
+                    Value__c = String.valueOf(i)
+            ));
+        }
+        insert requiredConfigs;
+
+        System.assertEquals(true, GE_PaymentServices.isElevateCustomer(),
```
Comment (ptewson-sfdo) [inline src/classes/GE_PaymentServices_TEST.cls:202]:
```suggestion
        System.assertEquals(true, GE_PaymentServices.isElevateCustomer,
```

File: src/classes/GE_PaymentServices_TEST.cls [hunk 3]
```diff
@@ -180,8 +180,35 @@ public class GE_PaymentServices_TEST {
...
+                'The org should be considered an Elevate customer when ' +
+                        'all required configuration keys are present.');
+
+        // Now delete a required configuration key and validate that
+        // GE_PaymentServices.isElevateCustomer returns false
+        delete [SELECT Id FROM Payment_Services_Configuration__c LIMIT 1];
+        System.assertEquals(false, GE_PaymentServices.isElevateCustomer(),
```
Comment (ptewson-sfdo) [inline src/classes/GE_PaymentServices_TEST.cls:209]:
```suggestion
        System.assertEquals(false, GE_PaymentServices.isElevateCustomer,
```

File: src/classes/GE_PaymentServices_TEST.cls [hunk 4]
```diff
@@ -180,8 +180,35 @@ public class GE_PaymentServices_TEST {
...
+            ));
+        }
+        insert requiredConfigs;
+
+        System.assertEquals(true, GE_PaymentServices.isElevateCustomer(),
+                'The org should be considered an Elevate customer when ' +
+                        'all required configuration keys are present.');
+
+        // Now delete a required configuration key and validate that
+        // GE_PaymentServices.isElevateCustomer returns false
```
Comment (ptewson-sfdo) [inline src/classes/GE_PaymentServices_TEST.cls:207]:
```suggestion
        // GE_PaymentServices.isElevateCustomer is false
```

**Pull request merged**